command-expansion-server/src/packages/lib/CommandTypeCodegen.ts

```diff
@@ -0,0 +1,216 @@
+    .join("")}] | [{\n${fswCommand.arguments
+    .map((argument) =>
+      argument.arg_type === "repeat" ? "" : `\t${argument.name}: ${mapArgumentType(argument, enumMap)},\n`
+    )
+    .join("")}}]): ${fswCommandName} {
+  return Command.new({
+    stem: '${fswCommand.stem}',
+    arguments: typeof args[0] === 'object' ? args[0] : args,
+  }) as ${fswCommandName};
+}`;
+
+  // language=TypeScript
+  const declaration = `
+${doc}
+\tfunction ${fswCommandName}(\n${fswCommand.arguments
+    .map((argument) =>
+      argument.arg_type === "repeat" ? "" : `\t\t${argument.name}: ${mapArgumentType(argument, enumMap)},\n`
+    )
+    .join("")}\t): ${fswCommandName};
+${doc}
+\tfunction ${fswCommandName}(args: {\n${fswCommand.arguments
+    .map((argument) =>
+      argument.arg_type === "repeat" ? "" : `\t\t${argument.name}: ${mapArgumentType(argument, enumMap)},\n`
+    )
+    .join("")}\t}): ${fswCommandName};
+\tinterface ${fswCommandName} extends Command<[\n${fswCommand.arguments
+    .map((argument) => (argument.arg_type === "repeat" ? "" : `\t\t${mapArgumentType(argument, enumMap)},\n`))
+    .join("")}\t] | {\n${fswCommand.arguments
+    .map((argument) =>
+      argument.arg_type === "repeat" ? "" : `\t\t${argument.name}: ${mapArgumentType(argument, enumMap)},\n`
+    )
+    .join("")}\t}> {}`;
+
+  return {
+    value,
+    declaration,
+  };
+}
+
+function mapArgumentType(argument: ampcs.FswCommandArgument, enumMap: ampcs.EnumMap): string {
+  switch (argument.arg_type) {
+    case "enum":
+      return enumMap[argument.enum_name].values.map((value) => `'${value.symbol}'`).join(" | ");
+    case "boolean":
+      return "boolean";
+    case "float":
+      return `F${argument.bit_length}`;
+    case "unsigned":
+      return `U${argument.bit_length}`;
+    case "integer":
+      return `I${argument.bit_length}`;
+    case "var_string":
+      return `VarString<${argument.prefix_bit_length}, ${argument.max_bit_length}>`;
+    case "fixed_string":
+      return "FixedString";
+    case "time":
+      return "string";
+    default:
+      throw new Error(`Unsupported argument type: ${argument.arg_type}`);
+  }
+}
+
+function versionBuilder(dictionary: ampcs.CommandDictionary) {
+  return `output/command_lib_${dictionary.header.mission_name.toLowerCase()}_${dictionary.header.version}.ts`;
+}
+
+export async function processDictionary(dictionary: ampcs.CommandDictionary) {
+  const prefaceUrl = new URL("../../../data/preface.txt", import.meta.url);
+  const folder = `${getEnv().STORAGE}/${dictionary.header.mission_name.toLowerCase()}/`;
+  const fileName = `command_lib.${dictionary.header.version}.ts`;
+  const prefaceString = await Promise.all([fs.promises.readFile(prefaceUrl.pathname, "utf8")]);
+
+  const { values, declarations } = generateTypescriptCode(dictionary); //update sql table store seprate
+
+  await fs.promises.mkdir(folder, { recursive: true });
+
+  return await Promise.resolve([
```

Comment:
promise.resolve not needed/undesirable here

Reply:
It was resovled fixed once I rebased with `develop`